muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/form-element.tsx

```diff
@@ -147,36 +146,57 @@ export class FormWizardActions extends React.Component<FormWizardActionsProps, F
 interface SearchFormElementProps {
   updateField: (value: string) => any,
   value?: string,
+  id: string,
   name: string,
-  placeholder?: string,
-  modifiers?: string | Array<string>
+  onFocus?: () => any,
+  onBlur?: () => any,
+  placeholder: string,
+  modifiers?: string | Array<string>,
+  className?: string,
 }
 
 interface SearchFormElementState {
-  value: string
+  clearActive: boolean
 }
 
 export class SearchFormElement extends React.Component<SearchFormElementProps, SearchFormElementState> {
 
   constructor(props: SearchFormElementProps) {
     super(props);
+    this.updateSearchField = this.updateSearchField.bind(this);
+    this.clearSearchField = this.clearSearchField.bind(this);
+
     this.state = {
-      value: this.props.value
+      clearActive: false
     }
-    this.updateSearchField = this.updateSearchField.bind(this);
   }
 
   updateSearchField(e: React.ChangeEvent<HTMLInputElement>) {
     let value = e.target.value;
+    if (e.target.value.length > 0) {
+      this.setState({ clearActive: true });
+    } else {
+      this.setState({ clearActive: false });
+    }
     this.props.updateField(value);
   }
 
+  clearSearchField() {
```

Comment:
I checked the codebase and it seems that the property value is never used and undefined is passed every time which means that the field is always uncontrolled, despite making it seem to be controlled.

Also the clearActive state serves no purpose, what I think should be done instead could be for example.

1. Remove clear active, and calculate whether the button should be active whether we have a value or not depending to what is provided.
2. Remove value from the props, and add it in the state, start it as an empty string.
3. Do not use this direct DOM manipulation, instead set the state as an empty string, never use direct dom manipulation with react components.

Reply:
Value is sometimes used, and sometimes not. Search component does not need a value, but can be given one -  if I recall correctly. I could make the value a mandatory field and fix the errors that come with it. I think the errors will only be in the organization searches.